phoenix-core/src/main/java/org/apache/phoenix/mapreduce/PhoenixInputFormat.java

```diff
@@ -87,74 +88,106 @@ public List<InputSplit> getSplits(JobContext context) throws IOException, Interr
-                }
+                        // The size is bogus, but it's not a problem
+                        psplits.add(new PhoenixInputSplit(Collections.singletonList(aScan),
+                                regionSize, regionLocation));
+                    }
                 } else {
-                if (LOGGER.isDebugEnabled()) {
-                    LOGGER.debug("Scan count[" + scans.size() + "] : " + Bytes.toStringBinary(scans
-                            .get(0).getStartRow()) + " ~ " + Bytes.toStringBinary(scans.get(scans
-                            .size() - 1).getStopRow()));
-                    LOGGER.debug("First scan : " + scans.get(0) + "with scanAttribute : " + scans
-                            .get(0).getAttributesMap() + " [scanCache, cacheBlock, scanBatch] : " +
-                            "[" + scans.get(0).getCaching() + ", " + scans.get(0).getCacheBlocks()
-                            + ", " + scans.get(0).getBatch() + "] and  regionLocation : " +
-                            regionLocation);
+                    if (LOGGER.isDebugEnabled()) {
+                        LOGGER.debug("Scan count[" + scans.size() + "] : "
+                                + Bytes.toStringBinary(scans.get(0).getStartRow()) + " ~ "
+                                + Bytes.toStringBinary(scans.get(scans.size() - 1).getStopRow()));
+                        LOGGER.debug("First scan : " + scans.get(0) + "with scanAttribute : "
+                                + scans.get(0).getAttributesMap()
+                                + " [scanCache, cacheBlock, scanBatch] : " + "["
+                                + scans.get(0).getCaching() + ", " + scans.get(0).getCacheBlocks()
+                                + ", " + scans.get(0).getBatch() + "] and  regionLocation : "
+                                + regionLocation);
 
-                    for (int i = 0, limit = scans.size(); i < limit; i++) {
-                        LOGGER.debug("EXPECTED_UPPER_REGION_KEY[" + i + "] : " + Bytes
-                                .toStringBinary(scans.get(i).getAttribute
-                                        (BaseScannerRegionObserver.EXPECTED_UPPER_REGION_KEY)));
+                        for (int i = 0, limit = scans.size(); i < limit; i++) {
+                            LOGGER.debug("EXPECTED_UPPER_REGION_KEY[" + i + "] : "
+                                    + Bytes.toStringBinary(scans.get(i).getAttribute(
+                                        BaseScannerRegionObserver.EXPECTED_UPPER_REGION_KEY)));
+                        }
                     }
+
+                    psplits.add(new PhoenixInputSplit(scans, regionSize, regionLocation));
                 }
+            }
 
-                psplits.add(new PhoenixInputSplit(scans, regionSize, regionLocation));
+            if (PhoenixConfigurationUtil.isMRRandomizeMapperExecutionOrder(config)) {
```

Comment:
This is the only code change, the rest of the changes in thsis method are just reformats